Code/Tools/SceneAPI/SceneData/Behaviors/ScriptProcessorRuleBehavior.cpp

```diff
@@ -108,24 +110,74 @@ namespace AZ::SceneAPI::Behaviors
             ExportProductList result;
             CallResult(result, FN_OnPrepareForExport, scene, outputDirectory, platformIdentifier, productList);
             ScriptBuildingNotificationBusHandler::BusDisconnect();
             return result;
         }
 
+        AZStd::atomic_int m_count = 1;
+
+        static BehaviorEBusHandler* Create()
+        {
+            return aznew ScriptBuildingNotificationBusHandler();
+        }
+
+        static void Destroy(BehaviorEBusHandler* behaviorEBusHandler)
+        {
+            auto* handler =
+                static_cast<ScriptBuildingNotificationBusHandler*>(behaviorEBusHandler);
+
+            --handler->m_count;
+            if (handler->m_count == 0)
+            {
+                delete handler;
+            }
+        }
+
         static void Reflect(AZ::ReflectContext* context)
         {
             if (AZ::BehaviorContext* behaviorContext = azrtti_cast<AZ::BehaviorContext*>(context))
             {
                 behaviorContext->EBus<ScriptBuildingNotificationBus>("ScriptBuildingNotificationBus")
-                    ->Attribute(AZ::Script::Attributes::Scope, AZ::Script::Attributes::ScopeFlags::Automation)
+                    ->Attribute(AZ::Script::Attributes::Scope, AZ::Script::Attributes::ScopeFlags::Common)
+                    ->Attribute(AZ::Script::Attributes::ExcludeFrom, AZ::Script::Attributes::ExcludeFlags::All)
                     ->Attribute(AZ::Script::Attributes::Module, "scene")
-                    ->Handler<ScriptBuildingNotificationBusHandler>()
+                    ->Handler<ScriptBuildingNotificationBusHandler>(&ScriptBuildingNotificationBusHandler::Create, &ScriptBuildingNotificationBusHandler::Destroy)
                     ->Event("OnUpdateManifest", &ScriptBuildingNotificationBus::Events::OnUpdateManifest)
                     ->Event("OnPrepareForExport", &ScriptBuildingNotificationBus::Events::OnPrepareForExport);
             }
         }
+
+        struct ScriptScope final
+        {
+            ScriptScope(ScriptBuildingNotificationBusHandler* self)
+                : m_self(self)
+            {
+                m_self->m_count++;
+            }
+
+            ~ScriptScope()
+            {
+                m_self->m_count--;
+                if (m_self->m_count == 0)
+                {
+                    // the script released the handler (i.e. set to None)
+                    BehaviorEBusHandler* self = m_self;
+                    AZStd::function<void()> destroySelf = [self]()
+                    {
+                        ScriptBuildingNotificationBusHandler::Destroy(self);
+                    };
+                    // Delay to delete self until the end of the scene pipeline
+                    m_self->m_count = 1;
+                    AZ::SceneAPI::Events::AssetPostImportRequestBus::QueueBroadcast(
+                        &AZ::SceneAPI::Events::AssetPostImportRequestBus::Events::CallAfterSceneExport, destroySelf);
+                }
```

Comment:
Mostly a nitpick but I think you can handle queuing without having to make a method on the bus by using QueueFunction: `AZ::SceneAPI::Events::AssetPostImportRequestBus::QueueFunction(&ScriptBuildingNotificationBusHandler::Destroy, m_self);`

Reply:
I can try that, but the lvalue was added do to a compile error, and frankly this allowed me to add a break point to see it working correctly.